cli/cmd/inject_util.go

```diff
@@ -139,6 +142,23 @@ func read(path string) ([]io.Reader, error) {
 	)
 	if path == "-" {
 		in = append(in, os.Stdin)
+	} else if isValidURL(path) {
+		var resp *http.Response
+		resp, err = http.Get(path)
+		if err != nil {
+			return nil, err
+		}
+
+		if resp.StatusCode == 404 {
```

Comment:
I think we should return an error if the status isn't a 200. See the similar code in kubectl [here](https://github.com/kubernetes/cli-runtime/blob/92e6ccd8a070991c43024c9ef9989b68b47452c1/pkg/resource/visitor.go#L280-L282).